lib/string_reverse.rb

```diff
@@ -1,6 +1,16 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string
+    length = my_string.length
+    (length / 2).times do |index|
+      temp = my_string[index]
+      my_string[index] = my_string[length - (index + 1)]
+      my_string[length - (index + 1)] = temp
+    end
+    return my_string
+  else
+    return nil

```

Comment:
I'd recommend exiting early rather than working through the entire method to return nil.